hypershift-operator/controllers/hostedcluster/internal/platform/agent/agent.go

```diff
@@ -45,13 +53,129 @@ func (p Agent) ReconcileCAPIInfraCR(ctx context.Context, c client.Client, create
 }
 
 func (p Agent) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, tokenMinterImage string) (*appsv1.DeploymentSpec, error) {
-	// It is expected that cluster-api-provider-agent and the Infrastructure Operator are installed as a prerequisite
-	return nil, nil
+	deploymentSpec := &appsv1.DeploymentSpec{
+		Replicas: k8sutilspointer.Int32Ptr(1),
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"control-plane": "controller-manager",
+				},
+			},
+			Spec: corev1.PodSpec{
+				TerminationGracePeriodSeconds: k8sutilspointer.Int64Ptr(10),
+				ServiceAccountName:            "cluster-api-provider-agent-controller-manager",
+				SecurityContext: &corev1.PodSecurityContext{
```

Comment:
Let's drop this since it introduce a userid assumption. OCP will use "restricted" scc by default automatically assigning a userid within a generated range.

https://docs.openshift.com/container-platform/4.9/authentication/managing-security-context-constraints.html#security-context-constraints-example_configuring-internal-oauth

https://cloud.redhat.com/blog/managing-sccs-in-openshift

Reply:
Done